publish/views.py

```diff
@@ -105,17 +116,29 @@ def create_route(request):
 
             ,
                 "purpose": request.POST.get('purpose'),
-                "s_point": request.POST.get('s_point'),
+                "s_point": request.POST.get('spoint'),
                 "destination": request.POST.get('destination'),
                 "type": request.POST.get('type'),
                 "date": request.POST.get("date"),
                 "hour": request.POST.get("hour"),
                 "minute":  request.POST.get("minute"),
                 "ampm": request.POST.get("ampm"),
-                "details": request.POST.get("details")
+                "details": request.POST.get("details"),
             }
         ride_id = request.POST.get('destination')
         attach_user_to_route(request.session['username'], route['_id'])
+        if(request.POST.get("slat")):
+            route["s_lat"] = request.POST.get("slat")
+            route["s_long"] = request.POST.get("slong")
+        if(request.POST.get("dlat")):
+            route["d_lat"] = request.POST.get("dlat")
+            route["d_long"] = request.POST.get("dlong")
+
+        if(request.POST.get("dlat") and request.POST.get("slat")):
+            res = mapsService.get_route_details(route["s_lat"], route["s_long"], route["d_lat"], route["d_long"])
```

Comment:
Add error handling if the get_route_details() API call cannot find a route between the source and destination.

Reply:
Added error handling !

Reply:
Add a timeout for the request

Reply:
Added!